compiler/luci/service/src/Nodes/CircleReshape.cpp

```diff
@@ -95,21 +97,13 @@ loco::TensorShape Algorithm::visit(const luci::CircleReshape *node)
     }
     else
     {
-      // We use shape from the node itself
-      loco::TensorShape shape;
-      shape.rank(node->rank());
-      for (uint32_t r = 0; r < node->rank(); ++r)
+      auto shape = loco::must_cast<luci::CircleNode *>(node->shape());
+      shape_by_input.rank(shape->dim(0).value());
```

Comment:
I don't understand this line. plz explain.

Reply:
If `node->shape()` is casted into `CircleNode`, `shape->rank()` always returns 1.

It's because, the `TensorShape` mixin only contains the information about the shape of the node itself.

For example, let's say some node (not `CircleConst`) has new shape info `[2, 3, 4]` in its value.
Then, the shape of the **node itself** is `[3]` which can be retrieved by `TensorShape`.
In that case, `rank` returns 1, and `dim(0)` return 3.

So we decided to return `[?, ?, ?]` in case of `CircleNode` because this is the only information that we can retrieve (`[3]`).

Related: https://github.com/Samsung/ONE/issues/13927#issuecomment-2328295368

Reply:
when `node->shape()` will is not a `CircleConst`, we want to use it's output tensor value
where this tensor should be rank 1, and from your example, value like `[2, 3, 4]`

when we call `shape->dim(0).value()`, this will be `3`, from `[2, 3, 4]`

As a code reader, variable names keeps confusion...

Reply:
> As a code reader, variable names keeps confusion...

Sorry for the confusion. I will take care to make variable names clear and understandable.

> when `node->shape()` will is not a `CircleConst`, we want to use it's output tensor value where this tensor should be rank 1, and from your example, value like `[2, 3, 4]`
> 
> when we call `shape->dim(0).value()`, this will be `3`, from `[2, 3, 4]`

For now, I'm afraid that there's no way to get the value `[2, 3, 4]` with `CircleNode` type.
It's because `CircleNode` is every node's base class and it doesn't inherits mixins that can retrieve that value(`[2, 3, 4]`).

I think this is current limitation and the `[?, ?, ?]` way is the best thing we can do for now, until new mixin that contains tensor's value introduced.

Reply:
> there's no way to get the value [2, 3, 4]

As I wrote, the "tensor"value, when we run this model from interpreter or runtime.
Just as an example to understand better.

Reply:
> As I wrote, the "tensor"value, when we run this model from interpreter or runtime. Just as an example to understand better.

Now I get it. I made some miscommunication.

I meant that there is no way to deliver the value of tensor with code because it cannot be accessed.